code-dot-org/code-dot-org
# P20 1134: Global Edition Configuration + Farsi Header Changes

This adds the ability to quickly create regional configurations.

## Region Configurations

Regions are defined as YAML files within `/config/global`. Simply adding a new file there will create a region with the same name as the filename. If we create, for instance, `/config/global/foo.yml`, we will now have a global region with the name `foo` accessible via `studio.code.org/global/foo`.

The configurations themselves are a simple key/value store. The initial depth of the configuration lists keys with standard names for different pages or contexts. For instance `header` is the header topbar and `footer` is the bottom navigation footer found on most pages.

Each of these can then have nested keys typically naming different components that exist within those contexts. The values for these can vary. A value of `false` will simply remove the component from that context. More complex options or further nesting may also exist. That depends on that component.

For instance, for the `header` context, there is a `teach-menu` context. This hosts nested keys for different links found in the "Teach" submenu that appears in the 'hamburger' menu on the header. We can disable individual links in there, or by setting `teach-menu` to `false` altogether (and not having any nested keys), we can remove all such links under that "Teach" submenu.

## Header Changes

Under the "Farsi" region, we have disabled the links and menus and such that are not visible in this region. We have done so via the `fa.yml` configuration and added rendering logic to the common header helpers (both Pegasus and Dashboard since they share their implementation).

`/home`: (as student)

![image](https://github.com/user-attachments/assets/abf5ac81-d004-4b37-a142-6522a91fb3ee)

`/global/fa/home`: (as student)

![image](https://github.com/user-attachments/assets/a46e0af8-46fd-4d5c-b225-de766e0f6504)

`/home`: (as student)

`/global/fa/home`: (as teacher)

![image](https://github.com/user-attachments/assets/8dc21c5c-2c50-425b-9018-006317ae30c8)

`/projects/<project id>`:

![image](https://github.com/user-attachments/assets/62af900c-49b9-4ed7-8921-6b1f68173c4d)

`/global/fa/projects/<project id>`: (Help menu truncated to not show links to documentation until we know it is translated and we can turn it back on for that region)

![image](https://github.com/user-attachments/assets/a72fe7d5-5d3a-454b-8ce9-bcefee3ee433)


## Links



- jira ticket: [P20-1134](https://codedotorg.atlassian.net/browse/P20-1134)
- jira ticket: [P20-1155](https://codedotorg.atlassian.net/browse/P20-1155)

## Testing

Manually compared the headers in all relevant situations:

* Logged in as teacher on code.org and studio.code.org
* Logged in as student on code.org and studio.code.org
* Not logged in on code.org and studio.code.org

In each scenario, I dragged the window size to each of the three widths our design supports to check for the collapsed links in the hamburger menus and their disappearance from the main header.

All links are the same and all behavior is the same.

UI tests also corroborate when they touch on the header / hamburger links.

## PR Checklist:
